pkg/util/pod/pod.go

```diff
@@ -78,3 +81,25 @@ func ExecCommand(cs *testclient.ClientSet, pod *corev1.Pod, command ...string) (
 
 	return buf.String(), errbuf.String(), nil
 }
+
+type Network struct {
+	Interface string
+	Addresses []string
+}
+
+// GetInterfaceAddress retrieves the ip addresses for the given interface on a pod
+func GetInterfaceAddress(cs *testclient.ClientSet, pod *corev1.Pod, intf string) ([]string, error) {
+	var nets []Network
+	err := json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/networks-status"]), &nets)
```

Comment:
Don't think so. If the record is not there, the zero value is an empty string.
It will result in the struct to be empty (as the fields are not required) but it won't crash.